### PR TITLE
[ONNX]Fix export of full_like

### PR DESCRIPTION
**Pull request by KsenijaS**
Fix export of full_like when fill_value is of type torch._C.Value.

This PR fixes a bug when exporting GPT2DoubleHeadsModel https://github.com/huggingface/transformers/issues/4950

### CODE REVIEW

**Comment by KsenijaS** [comment]
@neginraoof @BowenBao please review

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1512,13 +1512,16 @@ def full(g, sizes, value, dtype, layout, device, pin_memory=False):
...
 def full_like(g, input, fill_value, dtype=None, layout=None, device=None, pin_memory=False, memory_format=None):
-    shape = g.op("Shape", input)
-    if dtype is None:
-        dtype = 6  # float
-    return g.op("ConstantOfShape", shape,
-                value_t=torch.tensor([fill_value], dtype=sym_help.scalar_type_to_pytorch_type[dtype]))
+    if sym_help._is_value(fill_value):
+        dtype = 6 if dtype is None else dtype
+        tmp = zeros_like(g, input, dtype, layout, device)
+        return add(g, tmp, fill_value, g.op("Constant", value_t=torch.tensor(1)))
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset9.py:1519]: Why is the constant 1 alpha arg neede?

**Review by neginraoof — COMMENTED** [review]
Thanks! Can you add a test for this use-case of full_like?

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (KsenijaS) [reply torch/onnx/symbolic_opset9.py:1519]: if alpha == 1 add function will return onnx::Add op:
https://github.com/pytorch/pytorch/blob/master/torch/onnx/symbolic_opset9.py#L85

Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:1519]: What if alpha is None (not passed)? I don't think it's necessary to pass 1.

Comment (KsenijaS) [reply torch/onnx/symbolic_opset9.py:1519]: alpha cannot be passed to full_like. I agree it's not necessary to pass 1 to add op, but it's consistent with how add was used in other symbolic functions in the file.

Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:1519]: Actually this is not the case. Add() has been used without alpha in many instances. Still, this is not necessary.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -1512,13 +1512,16 @@ def full(g, sizes, value, dtype, layout, device, pin_memory=False):
                     value_t=torch.tensor([const_value], dtype=sym_help.scalar_type_to_pytorch_type[dtype]))
 
 
-@parse_args('v', 'f', 'i', 'v', 'v', 'v', 'v')
 def full_like(g, input, fill_value, dtype=None, layout=None, device=None, pin_memory=False, memory_format=None):
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset9.py:1515]: We would at least want two tests in onnx_onnxruntime to cover export for both branches of this functions.
One for sym_help._is_value(fill_value) branch and one for the other branch.

File: test/onnx/expect/TestOperators.test_full_like.expect [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ir_version: 6
 producer_name: "pytorch"
-producer_version: "XXX"
+producer_version: "1.7"
```
Comment (neginraoof) [inline test/onnx/expect/TestOperators.test_full_like.expect:3]: You don't need to include this file if the only change is the producer_version change from xxx to a version.
This allows these tests to compare model files ignoring of the producer_version.

**Review by neginraoof — APPROVED** [review]

**Comment by KsenijaS** [comment]
@houseroad can you please review? Thanks

**Comment by KsenijaS** [comment]
@houseroad can you please take a look? Thanks.

File: test/onnx/expect/TestOperators.test_full_like.expect (hunk 3 above)
Comment (houseroad) [inline test/onnx/expect/TestOperators.test_full_like.expect:3]: Is this expect file still used?

**Review by houseroad — APPROVED** [review]
LG, thanks

**Pull request closed without merging**